scrapy/pipelines/files.py

```diff
@@ -39,41 +41,40 @@ class FileException(Exception):
 
 
 class FSFilesStore:
-    def __init__(self, basedir):
+    def __init__(self, basedir: str):
         if '://' in basedir:
             basedir = basedir.split('://', 1)[1]
         self.basedir = basedir
-        self._mkdir(self.basedir)
-        self.created_directories = defaultdict(set)
+        self._mkdir(Path(self.basedir))
+        self.created_directories: defaultdict[str, set[str]] = defaultdict(set)
```

Comment:
If I'm not mistaken, this syntax is Python 3.9+

Reply:
You are right, but the tests somehow pass on 3.7? There are two instances of this in this file.

Reply:
This is not the first time I hear "this typing syntax shouldn't be supported on the oldest version but the tests pass for it". Not sure why is it so.

Reply:
I thought maybe there could be missing tests, but there are tests for that code, and they pass on 3.7 even if I put the annotations back to how they were. It works, even when not running the tests.

It turns out, Python 3.7 doesn't care what's written here at all. It doesn't need to be a real type.

```
Python 3.7.9 (tags/v3.7.9:13c94747c7, Aug 17 2020, 18:58:18) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> a: aoeaoeaoe = None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'aoeaoeaoe' is not defined
>>> class A:     
...     def __init__(self):
...         self.a: aoeaoeaoe = None
...
>>> A()
<__main__.A object at 0x0000019C495C8F08>
>>>
```